cedar-lean/Cedar/Validation/Typechecker.lean

```diff
@@ -185,13 +185,19 @@ def hasAttrInRecord (rty : RecordType) (x : Expr) (a : Attr) (c : Capabilities)
     else ok (.bool .anyBool) (Capabilities.singleton x a)
   | .none     => ok (.bool .ff)
 
+def actionType? (ety : EntityType) (acts: ActionSchema) : Bool :=
+  acts.keys.any (EntityUID.ty · == ety)
+
 def typeOfHasAttr (ty : CedarType) (x : Expr) (a : Attr) (c : Capabilities) (env : Environment) : ResultType :=
```

Comment:
curious why `typeOfHasAttr` needs adjusting, but no corresponding adjustment for `GetAttr`?

Reply:
Great question! The reason is that the Rust & Lean will both return an error if you try to access a nonexistent attribute. Currently, the Lean spec will return a `unknownEntity` error, while the Rust will return a different error (probably related to an unsafe attribute access). Since this is an error case for the Rust, no further information is checked about the Lean response.

Reply:
I think my question goes the other way.  If I have this straight

- the type of `Action::"view" has foo` clearly should be `False`.  This patch was required to get that behavior, because the existing code somehow wasn't already giving `False`.
- the type of `Action::"view".foo` clearly should be <error>.  No patch was required to get that behavior, because the existing code (correctly) was already giving an error.

So I guess the question is not "why didn't we need to adjust `GetAttr`" but "why does `HasAttr` need a special case for actions" given that the validator should already have a declaration for the type `Action` that involves no attributes

Reply:
Got it